File: frontend/src/api/menu.ts
import request from '@/utils/request'
import type { Result } from '@/types/api'
import type { MenuItem } from '@/types/menu'

export function getMenuTree(): Promise<Result<MenuItem[]>> {
  return request.get('/api/platform/menu/tree')
}

export function getMyMenus(): Promise<Result<MenuItem[]>> {
  return request.get('/api/platform/menu/my')
}

export function createMenu(data: { name: string; path?: string; icon?: string; parentId?: string; sortOrder?: number; status?: number; visible?: number }): Promise<Result<void>> {
  return request.post('/api/platform/menu/admin/create', data)
}

export function updateMenu(data: { id: string; name: string; path?: string; icon?: string; parentId?: string; sortOrder?: number; status?: number; visible?: number }): Promise<Result<void>> {
  return request.put('/api/platform/menu/admin/update', data)
}

export function deleteMenu(id: string): Promise<Result<void>> {
  return request.delete(`/api/platform/menu/admin/delete/${id}`)
}

File: backend/mrb-user/mrb-user-service/src/main/java/com/meetinghub/user/model/entity/Menu.java
package com.meetinghub.user.model.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.meetinghub.common.model.BaseEntity;
import lombok.Data;

/**
 * 菜单实体
 */
@Data
@TableName("menu")
public class Menu extends BaseEntity {

    private String name;

    private String path;

    private String icon;

    private Long parentId;

    private Integer sortOrder;

    private Integer visible;

    private Integer status;

}
